src/inc/quic_platform_winuser.h

```diff
@@ -776,13 +781,17 @@ QuicThreadCreate(
             ARRAYSIZE(WideName) - 1,
             Config->Name,
             _TRUNCATE);
+#if defined(QUIC_UWP_BUILD)
+        SetThreadDescription(*Thread, WideName);
+#else
         THREAD_NAME_INFORMATION ThreadNameInfo;
         RtlInitUnicodeString(&ThreadNameInfo.ThreadName, WideName);
         NtSetInformationThread(
```

Comment:
Do you get a build time or run time error from this?

Reply:
Build time. UWP has more locked down access to the Nt stuff. But at the same time, it's known to have SetThreadDescription in all cases.

Reply:
What is the build error exactly?

Reply:
Tons of undeclared identifier errors. The low level nt headers are just completely excluded from UWP. However, in regard to the discussion above, if we don't care about anything before Windows 10, we can just use SetThreadDescription for all windows builds.